logstash-core/lib/logstash/environment.rb

```diff
@@ -1,7 +1,40 @@
 # encoding: utf-8
 require "logstash/errors"
+require "logstash/config/cpu_core_strategy"
+require "logstash/settings"
 
 module LogStash
+
+  [
+            Setting::String.new("node.name", Socket.gethostname),
+           Setting::Boolean.new("config.allow_env", false),
+            Setting::String.new("config.path", nil, false),
+            Setting::String.new("config.string", nil, false),
+           Setting::Boolean.new("config.test", false),
+           Setting::Boolean.new("config.reload.auto", false),
+           Setting::Numeric.new("config.reload.interval", 3),
+           Setting::Boolean.new("metric.collect", true) {|v| v == true }, # metric collection cannot be disabled
+            Setting::String.new("settings.dir", ::File.join(Environment::LOGSTASH_HOME, "conf")),
+            Setting::String.new("pipeline.id", "main"),
+           Setting::Numeric.new("pipeline.workers", LogStash::Config::CpuCoreStrategy.maximum),
+           Setting::Numeric.new("pipeline.output.workers", 1),
+           Setting::Numeric.new("pipeline.batch.size", 125),
+           Setting::Numeric.new("pipeline.batch.delay", 5), # in milliseconds
+           Setting::Boolean.new("pipeline.unsafe_shutdown", false),
+                    Setting.new("plugin.paths", Array, []),
+            Setting::String.new("interactive", nil, false),
+           Setting::Boolean.new("debug", false),
+           Setting::Boolean.new("debug.config", false),
+           Setting::Boolean.new("verbose", false),
+           Setting::Boolean.new("quiet", false),
+           Setting::Boolean.new("version", false),
+           Setting::Boolean.new("help", false),
+            Setting::String.new("log.path", nil, false),
+           Setting::Boolean.new("log.json", false),
+            Setting::String.new("web_api.http.host", "127.0.0.1"),
+              Setting::Port.new("web_api.http.port", 9600),
+  ].each {|setting| SETTINGS.register(setting) }
```

Comment:
Might be not for now, but what do you think about having a less mixup default values registration? a few proposals:

* Why not having this in their own location, might be some place inside the _settings_ module?
* I guess in most of the cases we could infer the right type from the data type, most clear case is the boolean case. Might be we can avoid having to explicitly use this `Setting::Boolean` classes.

I ask specially the first one, because it might get simpler for maintenance and also when we might like to introduce somethings like validation for values, or what I do really find awesome, the `did you mean ....? ` followup.

What do you think?